src/main/OPBuilder.ts

```diff
@@ -1,33 +1,51 @@
 import { getUniResolver } from '@sphereon/did-uni-client';
+import { Format } from '@sphereon/pex-models';
 import { Resolvable, Resolver } from 'did-resolver';
 
 import { OP } from './OP';
 import { DIDJwt } from './functions';
 import { EcdsaSignature } from './types/JWT.types';
 import {
-  CredentialFormat,
   ExternalSignature,
   InternalSignature,
   PassBy,
+  ResponseIss,
   ResponseMode,
   ResponseRegistrationOpts,
+  ResponseType,
+  SigningAlgo,
+  SubjectType,
   SuppliedSignature,
 } from './types/SIOP.types';
 
 export default class OPBuilder {
-  didMethods: string[] = [];
+  expiresIn?: number;
+  idTokenSigningAlgValuesSupported: SigningAlgo[];
+  issuer: ResponseIss;
   resolvers: Map<string, Resolvable> = new Map<string, Resolvable>();
-  signatureType: InternalSignature | ExternalSignature | SuppliedSignature;
-  credentialFormats: CredentialFormat[] = [];
-  responseRegistration: ResponseRegistrationOpts;
   responseMode?: ResponseMode;
+  responseRegistration: ResponseRegistrationOpts;
+  responseTypesSupported: ResponseType[] | ResponseType;
   // did: string;
   // vp?: VerifiablePresentation;
-  expiresIn?: number;
   resolver?: Resolvable;
+  signatureType: InternalSignature | ExternalSignature | SuppliedSignature;
+  subjectSyntaxTypesSupported: string[] = [];
+  vpFormats: Format;
```

Comment:
Is vpFormats a required argument?

Reply:
yes it is https://openid.net/specs/openid-4-verifiable-presentations-1_0.html#section-8.2.3.1